Tests/SwiftFormatTests/PrettyPrint/Selections.swift

```diff
@@ -0,0 +1,231 @@
+import SwiftFormat
+import XCTest
+
+final class Selections: PrettyPrintTestCase {
+  func testSelectAll() {
+    let input =
+      """
+      ➡️func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+        // do stuff
+      }
+      }⬅️
+      """
+
+    let expected =
+      """
+      func foo() {
+        if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+          // do stuff
+        }
+      }
+      """
+
+    // The line length ends on the last paren of .Stuff()
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 80)
+  }
+
+  func testSelectComment() {
+    let input =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      ➡️// do stuff⬅️
+      }
+      }
+      """
+
+    let expected =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+          // do stuff
```

Comment:
Hmm, this is quite unfortunate. I would have preferred if we inferred the indentation from the unformatted context instead of from how the indentation should be if formatting was enabled.

FWIW running `clang-format` for lines 3-4 on 
```cpp
class Foo {
void test() {
int i = 0;
int j = 1;
}
}
```

produces

```cpp
class Foo {
void test() {
  int i = 0;
  int j = 1;
}
}
```

Reply:
Yeah, I find it unfortunate too. The engine here isn't really set up to start "in the middle" like this. But perhaps we can move it in that direction down the road.